source/common/http/http1/codec_impl.cc

```diff
@@ -936,6 +936,12 @@ Status ServerConnectionImpl::handlePath(RequestHeaderMap& headers, unsigned int
   // new Host field-value based on the received request-target rather than
   // forward the received Host field-value.
   headers.setHost(absolute_url.hostAndPort());
+  // Add the scheme. This will be validated in the HCM to ensure no https://
+  // requests are accepted over unencrypted connections by front-line Envoys.
```

Comment:
> Do we expect fully qualified URLs when xff_num_trusted_hops != 0?

I don't think so. I think if this is not an absolute URL we need to synthesize scheme depending on whether the underlying transport is TLS or not?

Reply:
Envoy never sends fully qualified urls, so in an envoy mesh no, but if num_trusted_hops > 0 and your trusted proxy is sending fully qualified
scheme_one://host urls with x-forwarded-proto scheme_two
that's kind of on you.  Right now scheme and x-forwarded proto are known to be inconsistent so we can not (yet) validate and reject if they're not the same.

I'm was the HCM set scheme when it's not present because we'd talked about moving more sanitization.
If scheme is not present, we want to set based on encryption *or* XFP, depending on if this is a frontline Envoy, and I don't think that logic belongs in the codec even if it's a problem specific to HTTP/1.1

Reply:
OK I see, so here we only set scheme in the absolute URL case, because otherwise we rely on the HCM to set XFP and then from that set scheme? Can you add some more comments?